### DISCUSSION TITLE
Is it possible to use "wire:init" for renders after the first render?

### DISCUSSION THREAD

**Discussion by JerrelZ** [livewire/livewire#5141] [1. Help]
Hi!

Using "wire:init" is great for minimising the load time of the initial render, which is great.

I was wondering if this is also possible for renders after the first render?

After the first render, the user on my website can change a lot of options on the page. When the user changes an option, the load time is a lot higher because I can't use "wire:init" anymore. 

Is there a workaround for this?

Thanks!

**Comment by Martleby** [comment]
I believe wire:init is a one time event when the component first appears on the page and doesn't run again.

However, there are a number of potential strategies you could use depending on what your component does, what you mean by slow, and why it is slow.

For example, I have a search form in my application and I only want this to run its search query when the user actually presses the search button, and not when the user changes various search parameters. Here's the approach I took:

```php

public $isSearching = false;

public function search()
{
    $this->isSearching = true;
}

public function getSearchResults()
{
   if(!$this->isSearching){
        return [];
    }

    $searchResults = /* Code to get the search results */;
    
    $this->isSearching = false;

    return $searchResults;
}

public function render()
{
   return view(''livewire.search-form')->with('data', $this->getSearchResults());
}

```

and then in the blade view I simply rig up a button with wire:click="search".

HTH,
Martin.

**Reply by JerrelZ** [reply]
Thanks Martin,

That's a clever way to solve this. I will try to implement something like this and report back to you.

**Comment by skalero01** [comment]
The idea of  [Martleby](https://github.com/Martleby) doesn't make sense, the idea of the defer is that if we are executing something that takes times we want to load it after the render so we dont need to wait all the time until all is loaded

i was able to do it by adding

`<div wire:key="{{ $render_number }}">{html}</div>` 

Putting the code with the wire:init inside this and changing the render_number on the component everytime a new search was done (Or some variables were changed)
